app/pages/project/project-page.jsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import Translate from 'react-translate-component';
+import { Markdown } from 'markdownz';
+import { sugarClient } from 'panoptes-client/lib/sugar';
+import ProjectNavbar from './project-navbar';
+import PotentialFieldGuide from './potential-field-guide';
+
+export default class ProjectPage extends React.Component {
+  constructor() {
+    super();
+    const lastSugarSubscribedID = null;
+  }
+
+  getChildContext() {
+    return this.context.geordi;
+  }
+
+  componentDidMount() {
+    document.documentElement.classList.add('on-project-page');
+    this.updateSugarSubscription(this.props.project);
+    (this.context.geordi) ? this.context.geordi.remember({ projectToken: this.props.project.slug }) : null;
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.project !== this.props.project) {
+      this.updateSugarSubscription(nextProps.project);
+      (this.context.geordi) ? this.context.geordi.remember({ projectToken: nextProps.project.slug }) : null;
+    }
+  }
+
+  componentWillUnmount() {
+    document.documentElement.classList.remove('on-project-page');
+    this.updateSugarSubscription(null);
+    this.context.geordi.forget(['projectToken']);
+  }
+
+  updateSugarSubscription(project) {
+    if (this.lastSugarSubscribedID) {
+      sugarClient.unsubscribeFrom(`project-${this.lastSugarSubscribedID}`);
+    }
+    if (project) {
+      sugarClient.subscribeTo(`project-${this.lastSugarSubscribedID}`);
+    }
+  }
+
+  render() {
+    const projectPath = `/projects/${this.props.project.slug}`;
+    const onHomePage = this.props.routes[2].path === undefined;
+    const pages = this.props.pages.reduce((map, page) => {
+      map[page.url_key] = page;
+      return map;
+    }, {});
+
+    const renderAnnouncement = (this.props.project.configuration && !!this.props.project.configuration.announcement) ?
```

Comment:
Small thing, but these are not functions so their names should be nouns and not verbs.

Reply:
Redundant double negative check here too, I think, but ESLint should catch that.